app/demo-loop/demo-loop.cc

```diff
@@ -79,19 +68,40 @@ void run(std::istream* is, std::shared_ptr<celeritas::OutputRegistry> output)
     // Create runner and save setup time
     celeritas::Stopwatch get_setup_time;
     Runner run_stream(*run_input, output);
-    double const setup_time = get_setup_time();
+    SimulationResult result;
+    result.setup_time = get_setup_time();
+    result.events.resize(run_stream.num_events());
 
-    // Run on a single thread
-    RunnerResult result = [&run_stream] {
+    celeritas::Stopwatch get_transport_time;
+    {
+        celeritas::MultiExceptionHandler capture_exception;
         celeritas::ScopedDeviceProfiling profile_this;
-        return run_stream(celeritas::StreamId{0});
-    }();
-    result.time.setup = setup_time;
 
-    // TODO: convert individual results into OutputInterface so we don't have
-    // to use this ugly "global" hack
-    output->insert(OutputInterfaceAdapter<RunnerResult>::from_rvalue_ref(
-        OutputInterface::Category::result, "*", std::move(result)));
+#ifdef _OPENMP
+        // Set the maximum number of nested parallel regions
+        // TODO: Enable nested OpenMP parallel regions for multithreaded CPU
+        // once the performance issues have been resolved. For now, limit the
+        // level of nesting to a single parallel region (over events) and
+        // deactivate any deeper nested parallel regions.
+        omp_set_max_active_levels(1);
+#endif
+#pragma omp parallel for num_threads(run_stream.num_streams())
+        for (size_type event = 0; event < run_stream.num_events(); ++event)
+        {
+#ifdef _OPENMP
+            size_type stream = omp_get_thread_num();
+#else
+            size_type stream = 0;
+#endif
```

Comment:
It might improve readability to have a reusable utility function `StreamId get_openmp_stream()` that wraps this?

Reply:
Just stuck it in demo-loop.cc, but if you can think of a better home for it I'll move it.